src/pool.spec.ts

```diff
@@ -217,6 +217,36 @@ describe("Generic pool", () => {
 });
 
 describe("Constant product pool", () => {
+  it("parsing state", async () => {
+    const testBed = await makeFreshTestBed({ poolType: "CONSTANT_PRODUCT" });
+    const pact = new PactClient(algod);
+
+    const pool = await pact.fetchPoolById(testBed.pool.appId);
+
+    expect(pool.primaryAsset.index).toBe(testBed.algo.index);
+    expect(pool.secondaryAsset.index).toBe(testBed.coin.index);
+
+    expect(pool.poolType).toBe("CONSTANT_PRODUCT");
+    expect(pool.version).toBe(2);
+
+    expect(pool.internalState).toEqual({
+      A: 0,
+      ADMIN: testBed.account.addr,
+      ASSET_A: pool.primaryAsset.index,
+      ASSET_B: pool.secondaryAsset.index,
+      LTID: pool.liquidityAsset.index,
+      B: 0,
+      CONTRACT_NAME: "PACT AMM",
+      FEE_BPS: pool.feeBps,
+      L: 0,
+      PACT_FEE_BPS: 0,
+      PRIMARY_FEES: 0,
+      SECONDARY_FEES: 0,
+      TREASURY: testBed.account.addr,
+      VERSION: 2,
+    });
```

Comment:
is it ok, that it ignores CONFIG?

Reply:
Yes, it's because CONFIG is decoded into ASSET_A, ASSET_B, and FEE_BPS and then removed from the state.